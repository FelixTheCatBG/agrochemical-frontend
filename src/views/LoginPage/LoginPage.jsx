import React from "react";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import { authenticationService } from "../../services";

import AgroLoader from "../../components/Shared/AgroLoader";
import AgroSnackbar from "../../components/Shared/AgroSnackbar";
import GridContainer from "../../components/Grid/GridContainer";

// function Copyright() {
//     return (
//         <Typography variant="body2" color="textSecondary" align="center">
//             {'Copyright Â© '}
//             <Link color="inherit" href="https://material-ui.com/">
//                 Your Website
//       </Link>{' '}
//             {new Date().getFullYear()}
//             {'.'}
//         </Typography>
//     );
// }


// const useStyles = makeStyles(theme => ({
//     paper: {
//         marginTop: theme.spacing(8),
//         display: "flex",
//         flexDirection: "column",
//         alignItems: "center"
//     },
//     avatar: {
//         margin: theme.spacing(1),
//         backgroundColor: theme.palette.secondary.main
//     },
//     form: {
//         width: "100%", // Fix IE 11 issue.
//         marginTop: theme.spacing(1)
//     },
//     submit: {
//         margin: theme.spacing(3, 0, 2)
//     }
// }));

const ErrorValidationLabel = ({ txtLbl }) => (
    <label htmlFor="" style={{ color: "red" }}>
        {txtLbl}
    </label>
);

class LoginPage extends React.Component {
    constructor (props) {
        super(props);

        this.state = {
            username: "",
            password: "",

            isLoading: false,
            success: false,
            action: '',
            errorMessage: false,
            successAction: "Login"
        };
    }

    handleKeyPress = e => {
        if (e.key === "Enter") {
            this.handleLogin();
        }
    };

    signIn = e => {
        e.preventDefault();

        this.setState({
            emailErrorMessage: "",
            passwordErrorMessage: ""
        });

        if (!/(.+)@(.+){2,}\.(.+){2,}/.test(this.state.username)) {
            this.setState({ emailErrorMessage: "Emails must be valid such as username@gmail.com" });

            return;
        }

        if (this.state.password.length < 4) {
            this.setState({ passwordErrorMessage: "Password must be more than 3 chars" });

            return;
        }

        const username = this.state.username;
        const password = this.state.password;

        authenticationService.login(username, password).then(user => {
            // console.log(user);
            const { from } = this.props.location.state || {
                from: { pathname: "/" }
            };

            this.props.history.push(from);
        }).catch(error => {
            return error;
        });
    };

    onSnackbarClose = () => {
        this.setState({
            successAction: ""
        });
    }

    handleInputChange = e => {
        const name = e.target.name;
        const value = e.target.value;

        this.setState({
            [name]: value
        });
    };


    render () {
        const disabled = !this.state.username || !this.state.password;

        const { emailErrorMessage, passwordErrorMessage } = this.state;

        if (this.state.isLoading) {
            return <GridContainer body>
                <AgroLoader />
            </GridContainer>;
        }

        return (
            <Container component="main" maxWidth="xs">
                <CssBaseline />
                <div style={{ marginTop: 50, marginBottom: 50 }}>
                    <Avatar>
                        <LockOutlinedIcon />
                    </Avatar>
                    <Typography component="h1" >
                        Sign in
                    </Typography>
                    <form>
                        <TextField
                            variant="outlined"
                            margin="normal"
                            required
                            fullWidth
                            id="email"
                            label="Email Address"
                            name="username"
                            autoComplete="email"
                            onChange={this.handleInputChange}
                            autoFocus
                        />

                        {emailErrorMessage && <ErrorValidationLabel txtLbl={emailErrorMessage} />}

                        <TextField
                            variant="outlined"
                            margin="normal"
                            required
                            fullWidth
                            name="password"
                            label="Password"
                            type="password"
                            id="password"
                            autoComplete="current-password"
                            onChange={this.handleInputChange}
                        />
                        {passwordErrorMessage && <ErrorValidationLabel txtLbl={passwordErrorMessage} />}

                        <FormControlLabel
                            control={<Checkbox value="remember" color="primary" />}
                            label="Remember me"
                        />
                        <Button
                            type="submit"
                            disabled={disabled}
                            fullWidth
                            variant="contained"
                            color="primary"
                            onClick={this.signIn}
                        >
                            Sign In
                        </Button>
                        <Grid container>
                            <Grid item xs>
                                <Link href="#" variant="body2">
                                    Forgot password?
                                </Link>
                            </Grid>
                            <Grid item>
                                <Link href="#" variant="body2">
                                    {"Don't have an account? Sign Up"}
                                </Link>
                            </Grid>
                        </Grid>
                    </form>
                </div>
                {this.state.successAction && <AgroSnackbar type={'success'} message={`${this.state.successAction} Successful`} onClose={this.onSnackbarClose} />}

                {this.state.errorMessage && <AgroSnackbar type={'error'} message={`Delete Successful`} onSnackbarClose={this.onSnackbarClose} />}
            </Container>
        );
    }
}

export default LoginPage;
