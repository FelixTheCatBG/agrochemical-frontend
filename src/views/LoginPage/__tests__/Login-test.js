// app/src/components/__tests__/Login-test.js
import React from 'react';
import { render } from '@testing-library/react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LoginPage from "../LoginPage";

configure({ adapter: new Adapter() });

describe('Login Component', () => {
    // it('should render without throwing an error', () => {
    //     expect(shallow(<LoginPage />).exists(<form></form>)).toBe(true);
    // });

    it('renders a email input', () => {
        expect(shallow(<LoginPage />).find('#email').length).toEqual(1);
    });

    it('renders a password input', () => {
        expect(shallow(<LoginPage />).find('#password').length).toEqual(1);
    });

    describe('Email input', () => {

        it('should respond to change event and change the state of the Login Component', () => {
            const wrapper = shallow(<LoginPage />);

            wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'blah@gmail.com' } });
            expect(wrapper.state('email')).toEqual('blah@gmail.com');
        });
    });

    describe('Password input', () => {

        it('should respond to change event and change the state of the Login Component', () => {
            const wrapper = shallow(<LoginPage />);

            wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'cats' } });
            expect(wrapper.state('password')).toEqual('cats');
        });
    });

    it('renders correctly', () => {
        const tree = render(<LoginPage />);

        expect(tree).toMatchSnapshot();
    });
});