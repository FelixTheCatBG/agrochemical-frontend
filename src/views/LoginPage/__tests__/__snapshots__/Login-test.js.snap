// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          style="margin-top: 50px; margin-bottom: 50px;"
        >
          <div
            class="MuiAvatar-root MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
              >
                <path
                  d="M0 0h24v24H0V0z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  opacity=".87"
                />
              </g>
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
          </div>
          <h1
            class="MuiTypography-root MuiTypography-body1"
          >
            Sign in
          </h1>
          <form>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
                data-shrink="true"
                for="email"
              >
                Email Address
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="email"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-109 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-110"
                    style="width: 0px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="password"
              >
                Password
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-109 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-110"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-123 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-126"
                    data-indeterminate="false"
                    type="checkbox"
                    value="remember"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Remember me
              </span>
            </label>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign In
              </span>
            </button>
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                  href="#"
                >
                  Forgot password?
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                  href="#"
                >
                  Don't have an account? Sign Up
                </a>
              </div>
            </div>
          </form>
        </div>
        <div
          class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft"
          data-qe="pop-up-error-message-box"
          variant="error"
        >
          <div
            aria-describedby="message-id"
            class="MuiTypography-root MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root AgroSnackbar-success-278 MuiTypography-body2"
            direction="up"
            role="alertdialog"
            style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="MuiSnackbarContent-message"
            >
              <span
                id="message-id"
              >
                Login Successful
              </span>
            </div>
            <div
              class="MuiSnackbarContent-action"
            >
              <button
                aria-label="Close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="MuiContainer-root MuiContainer-maxWidthXs"
    >
      <div
        style="margin-top: 50px; margin-bottom: 50px;"
      >
        <div
          class="MuiAvatar-root MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity=".87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-body1"
        >
          Sign in
        </h1>
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="email"
            >
              Email Address
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="username"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-109 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-110"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
            >
              Password
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-109 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-110"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-123 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-126"
                  data-indeterminate="false"
                  type="checkbox"
                  value="remember"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Remember me
            </span>
          </label>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign In
            </span>
          </button>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Forgot password?
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </form>
      </div>
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft"
        data-qe="pop-up-error-message-box"
        variant="error"
      >
        <div
          aria-describedby="message-id"
          class="MuiTypography-root MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root AgroSnackbar-success-278 MuiTypography-body2"
          direction="up"
          role="alertdialog"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiSnackbarContent-message"
          >
            <span
              id="message-id"
            >
              Login Successful
            </span>
          </div>
          <div
            class="MuiSnackbarContent-action"
          >
            <button
              aria-label="Close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
